surch/repo.py

```diff
@@ -192,31 +201,38 @@ def _get_user_details(self, sha):
             details, 'Date:   ', '+').strip()
         return name, email, commit_time
 
-    def search(self, search_list):
+    def search(self, search_list, commit_list=None):
```

Comment:
try to be more intent revealing when naming variables. quickly going over the code I'm not sure there's a real difference between 'commits' and 'commit_list'